generals/gui/gui.py

```diff
@@ -1,20 +1,50 @@
-from typing import Any
+import pygame
+from typing import Any, Literal
 
 from generals.core.game import Game
 from .properties import Properties
-from .event_handler import EventHandler
+from .event_handler import TrainEventHandler, GameEventHandler, ReplayEventHandler
 from .rendering import Renderer
 
 
 class GUI:
     def __init__(
-            self, game: Game, agent_data: dict[str, dict[str, Any]], from_replay=False
+        self,
+        game: Game,
+        agent_data: dict[str, dict[str, Any]],
+        mode: Literal["train", "game", "replay"] = "train",
     ):
-        self.properties = Properties(game, agent_data)
+        self.properties = Properties(game, agent_data, mode)
         self.__renderer = Renderer(self.properties)
-        self.__event_handler = EventHandler(self.properties, from_replay)
+        self.__event_handler = self.__initialize_event_handler()
+
+        pygame.init()
+        pygame.display.set_caption("Generals")
+
+        # Handle key repeats
+        pygame.key.set_repeat(500, 64)
+
+    def __initialize_event_handler(self):
+        if self.properties.mode == "train":
+            return TrainEventHandler
+        elif self.properties.mode == "game":
+            return GameEventHandler
+        elif self.properties.mode == "replay":
+            return ReplayEventHandler
 
     def tick(self, fps=None):
-        control_events = self.__event_handler.handle_events()
+        handler = self.__event_handler(self.properties)
+        command = handler.handle_events()
+        if command.quit:
+            quit()
```

Comment:
Won't this quit program before the integrations have a chance to call quit (and call `close`)?

Reply:
Yes, if close is called by pressing Q, its ok to `quit()`. Integrations should quit when called env.close programmatically

Reply:
Right, I missed that you removed checking for quit in integrations